adapta/storage/blob/s3_storage_client.py

```diff
@@ -100,9 +106,19 @@ def list_blobs(
         self, blob_path: DataPath, filter_predicate: Optional[Callable[[...], bool]] = None
     ) -> Iterator[DataPath]:
         """
-        Not implemented in S3 Client
+        Lists blobs in S3 storage.
+
+        :param blob_path: Path to blob
+        :param filter_predicate: Optional callable to filter blobs
+
+        :return: An iterator over a list of the blobs in the S3 storage
         """
-        raise NotImplementedError("Not implemented in S3StorageClient")
+        s3_path = cast_path(blob_path)
+        response = self._s3_resource.meta.client.list_objects(Bucket=s3_path.bucket, Prefix=s3_path.path, Delimiter="/")
+        if "Contents" in response:
```

Comment:
since you declare that you return an Iterator, you should return an empty iterator in case this condition is false